src/python/WMCore/Storage/Backends/GFAL2Impl.py

```diff
@@ -28,17 +28,35 @@ def __init__(self, stagein=False):
         self.removeCommand = self.setups % '. $JOBSTARTDIR/startup_environment.sh; printenv; date; gfal-rm -vvv -t 600 %s '
         self.copyCommand = self.setups % '. $JOBSTARTDIR/startup_environment.sh; printenv; date; gfal-copy -vvv -t 2400 -T 2400 -p %(checksum)s %(options)s %(source)s %(destination)s'
 
-    def createSourceName(self, protocol, pfn):
+
+    def createFinalPFN(self, protocol, pfn):
         """
-        _createSourceName_
-        GFAL2 uses file:/// urls
+        _createFinalPFN_
+        GFAL2 requires file:/// for any direction transfers
         """
         if pfn.startswith('file:'):
             return pfn
         elif os.path.isfile(pfn):
```

Comment:
What you mean by remote file? with hostname or only /store/mc/... ?
Not the same.
```
os.path.isfile(path)
Return True if path is an existing regular file. This follows symbolic links, so both islink() and isfile() can be true for the same path.
```
'/' needed if you would do gfal-copy srm://....file.root /store/user/... in that case the destination file does not exist and you need to append file:// in beginning. isfile will be false in this case, but '/' not and will add file:// 

In the end, any source or destination file which starts with:
srm:// or gridftp:// or anyOtherProcol:// will be returned as it is
file: will be returned as it is
fileName.root will return '"file://%s"  % abspath(fileName.root)
/store/fileName.root will return 'file:///store/fileName.root'

and what LogCollect does it sets destination like that:
```
gfal-copy -vvv -t 2400 -T 2400 -p   gsiftp://red-gridftp.unl.edu//mnt/hadoop/user/uscms01/pnfs/unl.edu/data4/cms/store/unmerged/logs/prod/2016/6/30/pdmvserv_task_HIG-RunIISpring16DR80-01259__v1_T_160620_125903_751/HIG-RunIISpring16DR80-01259_0/0012/0/54670436-3c68-11e6-900f-001e67abefa8-19-0-logArchive.tar.gz /var/lib/condor/execute/dir_2916738/glide_2t8wVK/execute/dir_40989/job/WMTaskSpace/logCollect1/54670436-3c68-11e6-900f-001e67abefa8-19-0-logArchive.tar.gz
```
So now it is not correct and with this fix it will have file:// in beginning of dest file.